src/modules/commander/Arming/PreFlightCheck/checks/preArmCheck.cpp

```diff
@@ -162,7 +163,7 @@ bool PreFlightCheck::preArmCheck(orb_advert_t *mavlink_log_pub, const vehicle_st
 
 	// Arm Requirements: authorization
 	// check last, and only if everything else has passed
-	if (arm_requirements.arm_authorization && prearm_ok) {
+	if (arm_requirements.arm_authorization && prearm_ok && real_arm_request) {
```

Comment:
Can you add `real_arm_request` to the vehicle command?

Reply:
I don't understand what you mean by 'vehicle command'.
May be you propose to add it to `arm_requirements`but I think it do not fit well there.

Reply:
I meant to add it to https://github.com/PX4/PX4-Autopilot/blob/813aff74fba6df362fa530a9f8016d22e73ba89f/src/modules/commander/Arming/ArmAuthorization/ArmAuthorization.cpp#L160, as `param1`, so the other side can decide whether to do the expensive check or not.

Reply:
I am not sure that's a good thing. An authorizer should not allow arming without doing the needed check.
More over, if we do that, PX4 would have to know if the stored authorization was just for check or real authorization...

For now, I removed the concerned commit for now. If we don't find a good way to do that, other commits can be merged while another solution is developed for mavlink log and authorizer spamming.

Reply:
I see your point. Ok let's keep it as is for now.

> More over, if we do that, PX4 would have to know if the stored authorization was just for check or real authorization...

Correct.